components/derive/src/service_factory.rs

```diff
@@ -120,15 +113,20 @@ impl ToTokens for ServiceFactory {
     fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
         let name = &self.ident;
         let cr = &self.cr;
-        let artifact_id = self.artifact_id();
+        let artifact_name = self.artifact_name();
+        let artifact_version = self.artifact_version();
         let artifact_protobuf_spec = self.artifact_protobuf_spec();
         let service_constructor = self.service_constructor();
         let (impl_generics, ty_generics, where_clause) = self.generics.split_for_impl();
 
         let expanded = quote! {
             impl #impl_generics #cr::runtime::rust::ServiceFactory for #name #ty_generics #where_clause {
-                fn artifact_id(&self) -> #cr::runtime::rust::RustArtifactId {
-                    #artifact_id.parse().unwrap()
+                fn artifact_id(&self) -> #cr::runtime::ArtifactId {
+                    #cr::runtime::ArtifactId {
+                        runtime_id: 0,
```

Comment:
Nit: I think it's better to use a dedicated constant here: `#cr::runtime::RuntimeIdentifier::Rust as u32`.